### PR TITLE
docs: update CI build badages

### LINKED ISSUE #40325: Outdated CI badges and triggers in documents
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
No need

### 2. What did you expect to see? (Required)
Correct CI badges and documents.
### 3. What did you see instead (Required)
- 2 CI badges are outdated, had no builds for a long time.
- Triggers in file `ci.md` are outdated.
 
### 4. What is your TiDB version? (Required)
No need

### PR DESCRIPTION
**Pull request by wuhuizuo**
### What problem does this PR solve?

Issue Number: close #40325

Problem Summary: Outdated CI badges and triggers in documents

### What is changed and how it works?

- delete the deprecated `CircleCI` badage in README.md
- replace `Build Status` with Prow bot's badage in README.md
- delete outdated and useless file ci.md

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ci.md [hunk 1]
```diff
@@ -1,31 +0,0 @@
-# Commands to trigger ci pipeline
```
Comment (hawkingrei) [inline ci.md:1]: Do you have documentation for Prow?

Comment (wuhuizuo) [reply ci.md:1]: No need special document, now easy to trigger them.
Updated dev guide: https://pingcap.github.io/tidb-dev-guide/get-started/write-and-run-unit-tests.html#running-tests-for-a-pull-request

**Review by xhebox — APPROVED** [review]

**Review by purelind — APPROVED** [review]

File: ci.md (hunk 1 above)
Comment (hawkingrei) [reply ci.md:1]: Have you documentation about how to manage the pipeline and add a new job into the pipeline?

File: README.md [hunk 2]
```diff
@@ -2,11 +2,10 @@
 
 [![LICENSE](https://img.shields.io/github/license/pingcap/tidb.svg)](https://github.com/pingcap/tidb/blob/master/LICENSE)
 [![Language](https://img.shields.io/badge/Language-Go-blue.svg)](https://golang.org/)
-[![Build Status](https://travis-ci.org/pingcap/tidb.svg?branch=master)](https://travis-ci.org/pingcap/tidb)
+[![Build Status](https://prow.tidb.net//badge.svg?jobs=pingcap/tidb/merged_*)](https://prow.tidb.net/?repo=pingcap%2Ftidb&type=postsubmit)
```
Comment (hawkingrei) [inline README.md:5]:
```suggestion
[![Build Status](https://prow.tidb.net/badge.svg?jobs=pingcap/tidb/merged_*)](https://prow.tidb.net/?repo=pingcap%2Ftidb&type=postsubmit)
```

File: ci.md (hunk 1 above)
Comment (wuhuizuo) [reply ci.md:1]: I will add later when post submit pipelines are refactored.

Comment (wuhuizuo) [reply ci.md:1]: Current pinned a guide issue [#pingcap-qe/ci/](https://github.com/PingCAP-QE/ci/issues/1693). I plan to transform it to `tidb-dev-guide` repo

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**